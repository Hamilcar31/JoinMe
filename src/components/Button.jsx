import React from 'react';
import {
  Text, View, StyleSheet, TouchableOpacity,
} from 'react-native';

import {Strokes, Shadows} from '../constants';

const styles = StyleSheet.create({
  button: {
    height: 42,
    width: 160,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#2F74C5',
    shadowColor: '#000',
    margin: 10,
    ...Strokes,
    ...Shadows,
  },
  text: {
    color: 'white',
  },
});

export default function Button(props) {

  const {style = {}, textStyle = {}, onPress} = props;

  return (
    <View style={[styles.button, style]}>
      <TouchableOpacity onPress={onPress}>
        <Text style={[styles.text, textStyle]}>{props.title}</Text>
      </TouchableOpacity>
    </View>
  );
}

