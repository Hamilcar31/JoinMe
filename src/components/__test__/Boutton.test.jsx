import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../Button';

describe('Button component', () => {
  it('has a title', () => {
    // given
    const button = <Button title="title" />;

    // when
    const tree = renderer.create(button).toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });
});
