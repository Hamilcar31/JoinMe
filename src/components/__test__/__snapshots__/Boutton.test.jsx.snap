// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component has a title 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2F74C5",
      "borderColor": "#2F74C5",
      "borderRadius": 20,
      "borderWidth": 1,
      "elevation": 5,
      "height": 42,
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
      "width": 160,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
        },
        Object {},
      ]
    }
  >
    title
  </Text>
</View>
`;
